### PR TITLE
Joi creation

### PR DESCRIPTION
**Pull request by jrans**
## THIS PR
- For a given config and array of fields with extra options if needed create a joi schema object to be used later for validation
- Note will implement all methods in joi if shallow
- No requirements on the object keys just yet but can add that feature later
- Also if we can store nested objects may want to make possibel to have nested schema but keeping ti simple for now
  Related #45

**Certainly something to be iterated on and add to once we start using!**

### CODE REVIEW

File: test/config_validator.test.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  );
+  t.throws(
+    validator({ table_name: 'test' }),
+    'error if no fields property'
+  );
+  t.throws(
+    validator({ table_name: '2test', fields: {} }),
+    'error if table name doesn\t pass db name regex'
+  );
+  t.throws(
```
Comment (eliasmalik) [inline test/config_validator.test.js]: duplicate assertion?

**Review by eliasmalik — CHANGES_REQUESTED** [review]

File: test/config_validator.test.js (hunk 1 above)
Comment (jrans) [reply test/config_validator.test.js]: ahh the problem with copy and paste

**Review by eliasmalik — APPROVED** [review]

**Comment by nelsonic** [comment]
@jrans would you mind rebasing this branch on `master` to reduce the _diff_ for review? (_please/thanks_!)

**Comment by jrans** [comment]
@nelsonic Of course! Did it before I even saw the comment 😉

**Comment by nelsonic** [comment]
Thanks @jrans 🎉 
applause

**Pull request merged**